pkg/vpcmodel/drawioOutput.go

```diff
@@ -132,13 +132,26 @@ func (d *DrawioOutputFormatter) createRouters() {
 }
 
 func (d *DrawioOutputFormatter) lineRouter(line *groupedConnLine, vpcResourceName string) drawio.IconTreeNodeInterface {
-	switch {
-	case d.cConfigs[vpcResourceName].IsMultipleVPCsConfig:
+	if d.cConfigs[vpcResourceName].IsMultipleVPCsConfig {
 		return d.multiVpcRouters[vpcResourceName]
+	}
+	var routeredEP EndpointElem
+	switch {
 	case line.dst.IsExternal():
-		return d.nodeRouters[d.gen.TreeNode(line.src)]
+		routeredEP = line.src
 	case line.src.IsExternal():
-		return d.nodeRouters[d.gen.TreeNode(line.dst)]
+		routeredEP = line.dst
+	default:
+		return nil
+	}
+	if group, ok := routeredEP.(*groupedEndpointsElems); ok {
+		firstRouter := d.nodeRouters[d.gen.TreeNode((*group)[0])]
+		for _, node := range *group {
+			if d.nodeRouters[d.gen.TreeNode(node)] != firstRouter {
```

Comment:
Note that we can add, optionally with a bool flag,  the router to the grouping key, in which case we will not have to return nil in this case. Returning nil when a Router exists can be misleading, I guess.

Reply:
not sure I understand